presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SetOperationNodeUtils.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.optimizations;
+
+import com.google.common.collect.ImmutableListMultimap;
+import com.google.common.collect.ListMultimap;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+public class SetOperationNodeUtils
+{
+    private SetOperationNodeUtils() {}
+
+    public static <K, V> ListMultimap<K, V> toListMultimap(LinkedHashMap<K, List<V>> map)
```

Comment:
These functions are so generic it's weird to call the class `SetOperationNodeUtils` though that's where they are used.

Reply:
I wonder if we should call this class simply `LinkedHashMaps`? The methods are fairly generic, and not specific to Set operations